stablehlo/tests/interpret_sign.mlir

```diff
@@ -0,0 +1,43 @@
+// RUN: stablehlo-translate --interpret -split-input-file %s
```

Comment:
We recently migrated our testing infrastructure from lit-based testing. Here is the file following our current format (also removed duplicate comment under complex test):
```
// RUN: stablehlo-translate --interpret -split-input-file %s

// CHECK-LABEL: Evaluated results of function: sign_op_test_si64
func.func @sign_op_test_si64() {
  %operand = stablehlo.constant dense<[-1, 0, 1]> : tensor<3xi64>
  %result = stablehlo.sign %operand : tensor<3xi64>
  check.expect_eq_const %result, dense<[-1, 0, 1]> : tensor<3xi64>
  func.return
}

// -----

// CHECK-LABEL: Evaluated results of function: sign_op_test_f64
func.func @sign_op_test_f64() {
  // +NaN, -1.0, -0.0, +0.0, 1.0
  %operand = stablehlo.constant dense<[0x7FFFFFFFFFFFFFFF, -1.0, -0.0, 0.0, 1.0]> : tensor<5xf64>
  %result = stablehlo.sign %operand : tensor<5xf64>
  check.expect_almost_eq_const %result, dense<[0x7FFFFFFFFFFFFFFF, -1.0, -0.0, 0.0, 1.0]> : tensor<5xf64>
  func.return
}

// -----

// CHECK-LABEL: Evaluated results of function: sign_op_test_c128
func.func @sign_op_test_c128() {
  // (+NaN, +0.0), (+0.0, +NaN), (0.0, 1.0)
  %operand = stablehlo.constant dense<[(0x7FF0000000000001, 0x0000000000000000),
                                       (0x0000000000000000, 0x7FF0000000000001),
                                       (0.0, 1.0)]> : tensor<3xcomplex<f64>>
  %result = stablehlo.sign %operand : tensor<3xcomplex<f64>>
  check.expect_almost_eq_const %result, dense<[(0x7FF0000000000001, 0x7FF0000000000001),
                                               (0x7FF0000000000001, 0x7FF0000000000001),
                                               (0.0, 1.0)]> : tensor<3xcomplex<f64>>
  func.return
}
```

Reply:
Hello @ghpvnist ,

Shouldn't the output for `sign_op_test_c128()` be the following:

```mlir
dense<[
    (0x7FF0000000000001, 0.0),
    (0x7FF0000000000001, 0.0),
    (0.0, 1.0)
]> : tensor<3xcomplex<f64>>
```

Wouldn't the imaginary part be zero ?

Reply:
Hi @karthikrangasai, as explained in my earlier comment, HLO also returns NaNs for both real and imaginary parts. As StableHLO aims to follow closely with HLO, we should also do the same here.
For complex types, abs(x) returns a float, so the actual computation is like:
```
x / (abs(x), abs(x)) =  (x.real / NaN, x.imag / NaN) = (NaN, NaN)
```

Reply:
Thanks a lot @ghpvnist .

Reply:
This conversation has been resolved, but I don't think the code has been updated?

Reply:
Oh sorry, I thought I pushed the changes. Pushing them now.